backend/inspirehep/records/tasks.py

```diff
@@ -96,10 +99,14 @@ def update_references_pointing_to_merged_record(
             referenced_records_in_path = flatten_list(
                 get_value(matched_inspire_record, path[: -len(".$ref")], [])
             )
+
             for referenced_record in referenced_records_in_path:
-                update_reference_if_reference_uri_matches(
-                    referenced_record, merged_record_uri, new_record_uri
-                )
+                update_reference_if_reference_uri_matches(referenced_record, merged_record_uri, new_record_uri)
+            referenced_records_in_path = dedupe_list_of_dicts(referenced_records_in_path)
```

Comment:
I think it's not needed anymore with the logic inside `remove_duplicate_refs_from_record`, you iterate through the list of refs twice then